Source/WebCore/animation/WebAnimation.cpp

```diff
@@ -1923,11 +1923,23 @@ void WebAnimation::setRange(TimelineRange range)
         effect->animationRangeDidChange();
 }
 
+const TimelineRange& WebAnimation::range()
+{
+    RefPtr keyframeEffect = dynamicDowncast<KeyframeEffect>(m_effect);
+
+    if (m_specifiedRangeStart && keyframeEffect)
+        m_timelineRange.start = SingleTimelineRange::range(*m_specifiedRangeStart, SingleTimelineRange::Type::Start, nullptr, keyframeEffect->target());
+    if (m_specifiedRangeEnd && keyframeEffect)
+        m_timelineRange.end = SingleTimelineRange::range(*m_specifiedRangeEnd, SingleTimelineRange::Type::End, nullptr, keyframeEffect->target());
```

Comment:
You should wrap this all up in a `if (RefPtr keyframeEffect = dynamicDowncast<KeyframeEffect>(m_effect))` statement.

Reply:
Fixed.